__tests__/integration/query-circ-check-barcode.test.ts

```diff
@@ -0,0 +1,66 @@
+import { dataSources, queryTestServer } from '../setupJest';
+import assert from 'assert';
+
+it('resolves items', async () => {
+
+    // Source: https://github.com/sul-dlss/sul-requests/blob/35ad5eb23429c838289520eae7a1e91614cf33ae/app/services/folio_graphql_client.rb#L48
+    // in sul-requests `barcode` is passed in directly as a string; replicating that here by hardcoding it below
+    const query = `query CircCheckByBarcode {
+        items(barcode: "10050402-3001") {
+            status {
+                name
+            }
+            barcode
+            dueDate
+            effectiveCallNumberComponents {
+                callNumber
+            }
+            instance {
+                title
+            }
+        }
+    }`;
+
+    // mock the check on loans for this item
+    dataSources.circulation.getLoans = jest.fn().mockResolvedValue([]);
+
+    // mock the resovler for the instance
+    dataSources.instances.getByHoldingsRecordId = jest.fn().mockResolvedValue(
+        {
+            id: '1dd708e8-2a1f-5a18-878d-7c9ff82dae9f',
+            hrid: 'a10050402',
+            source: 'MARC',
+            title: 'Annual report.',
+            instanceTypeId: '30fffe0e-e985-4144-b2e2-1e8179bdb41f',
+            physicalDescriptions: ['v. : ill. ; 28 cm.'],
+            languages: ['eng'],
+        }
+    );
+
```

Comment:
Assume that this method is needed because this is how the connected instance for the item could be retrieved? Just trying to understand why these specific functions have been mocked. thanks!

Reply:
Thank you! That makes a lot of sense.